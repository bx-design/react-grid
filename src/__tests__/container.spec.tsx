import { render, screen } from '@testing-library/react'
import Container from '../container'
import type { ContainerProps } from '../types'

const customRender = (props?: ContainerProps) => {
  return render(<Container {...props} />)
}

describe('Container', () => {
  test('render Container component', () => {
    customRender()
    const container = screen.getByTestId('container')
    expect(container).toBeInTheDocument()
    expect(container).toHaveClass('container')
    expect(container).toMatchSnapshot()
  })

  test('render Container with fluid prop', () => {
    customRender({ fluid: true })
    const container = screen.getByTestId('container')
    expect(container).toBeInTheDocument()
    expect(container).toHaveClass('container-fluid')
    expect(container).toMatchSnapshot()
  })

  test('render Container with breakpoints props', () => {
    customRender({ sm: true, md: true, lg: true, xl: true, xxl: true })
    const container = screen.getByTestId('container')
    expect(container).toBeInTheDocument()
    expect(container).toHaveClass('container-sm')
    expect(container).toHaveClass('container-md')
    expect(container).toHaveClass('container-lg')
    expect(container).toHaveClass('container-xl')
    expect(container).toHaveClass('container-xxl')
    expect(container).toMatchSnapshot()
  })
})
