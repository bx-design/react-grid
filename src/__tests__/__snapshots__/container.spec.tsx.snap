// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container render Container component 1`] = `
<div
  class="container"
  data-testid="container"
/>
`;

exports[`Container render Container with breakpoints props 1`] = `
<div
  class="container container-sm container-md container-lg container-xl container-xxl"
  data-testid="container"
/>
`;

exports[`Container render Container with fluid prop 1`] = `
<div
  class="container-fluid"
  data-testid="container"
/>
`;
